evm/src/NttManager/NttManager.sol

```diff
@@ -543,6 +571,10 @@ contract NttManager is INttManager, RateLimiter, ManagerBase {
         return seq;
     }
 
+    /// @dev Override this function to provide an additional payload on the NativeTokenTransfer
+    /// @return - The bytes to be sent as the additional payload
+    function _prepareAdditionalPayload() internal pure virtual returns (bytes memory) {}
```

Comment:
I think you mentioned that no transfer context is needed to prepare the payload for the first use case of this feature, but does it make sense to pass in the key user provided info like `amount`, `recipient` and `recipientChain` as args here for the more general use case?

Reply:
I agree, will update.

Reply:
Updated the function to return the NTT struct which _might_ be more effective? lmk your thoughts

Reply:
Nice idea, that feels a bit cleaner to me